Cabal/tests/PackageTests/Tests.hs

```diff
@@ -120,10 +120,6 @@ tests config = do
   -- with the same name and LATER version
   tc "BuildDeps/InternalLibrary3" $ internal_lib_test "internal"
 
-  -- Test that an explicit dependency constraint which doesn't
```

Comment:
On a second thought, I think we should leave that but change it to check that it fails properly when using the `Setup.hs` interface.